src/Components/Web.JS/src/Platform/Mono/MonoPlatform.ts

```diff
@@ -249,16 +249,38 @@ async function createEmscriptenModuleInstance(resourceLoader: WebAssemblyResourc
   const existingPostRun = moduleConfig.postRun || [];
   (moduleConfig as any).preloadPlugins = [];
 
+  var totalResources = Object.keys(resources.assembly).length + Object.keys(resources.pdb || {}).length + /* dotnet.wasm */ 1;
+
+  if (resourceLoader.bootConfig.resources.runtime.hasOwnProperty('dotnet.timezones.blat')){
+    totalResources++;
+  }
+
+  if (resourceLoader.bootConfig.icuDataMode !== ICUDataMode.Invariant){
+    totalResources++;
+  }
+
+  let resourcesLoaded = 0;
+  function setProgress(){
+      resourcesLoaded++;
+      const percentage = resourcesLoaded / totalResources * 100;
+      document.documentElement.style.setProperty('--blazor-load-percentage', percentage + '%');
+      document.documentElement.style.setProperty('--blazor-load-percentage-text', '"' + Math.round(percentage) + '%"');
```

Comment:
nit; this looks like a good candidate for string interpolation:

```suggestion
      document.documentElement.style.setProperty('--blazor-load-percentage', `${percentage}%`);
      document.documentElement.style.setProperty('--blazor-load-percentage-text', `${Math.round(percentage)}%`);
```